jina/docker/hubio.py

```diff
@@ -225,7 +264,7 @@ def build(self) -> Dict:
                         self.logger.error(f'can not use it in the Flow')
                         is_build_success = False
 
-                if self.args.push:
+                if is_build_success and self.args.push:
                     try:
                         self.push(image.tags[0], self.readme_path)
```

Comment:
For the `build --push` case, `self.push()` function will search for temp file to be present in the disk and hence fail. We cannot write to the file before push, as we keep push info in summary too. 
We need to call `self._push_docker_hub()` here & do the file write + db write in the same function.

Reply:
This is done & tested.